h2/src/test/org/h2/test/jdbc/TestPreparedStatement.java

```diff
@@ -458,6 +459,28 @@ private void testUUID(Connection conn) throws SQLException {
         stat.execute("drop table test_uuid");
     }
 
+    private void testUUIDAsJavaObject(Connection conn) throws SQLException {
+        String uuidStr = "12345678-1234-4321-8765-123456789012";
+
+        Statement stat = conn.createStatement();
+        stat.execute("create table test_uuid(id uuid primary key)");
+        UUID origUUID = UUID.fromString(uuidStr);
+        PreparedStatement prep = conn.prepareStatement("insert into test_uuid values(?)");
+        prep.setObject(1, origUUID, java.sql.Types.JAVA_OBJECT);
+        prep.execute();
+
+        prep = conn.prepareStatement("select * from test_uuid where id=?");
+        prep.setObject(1, origUUID);
```

Comment:
No, I have no doubts that selecting by by UUID works fine in H2, I meant what if it will be `setObject(1, origUUID, Types.JAVA_OBJECT)` or `setObject(1, origUUID, Types.OTHER)`?